cmd/cmd/apiDelete.go

```diff
@@ -0,0 +1,146 @@
+
+	apiDeleteCmd.Flags().StringVarP(&apiToDelete, "name", "n", "", "Name of the API to be deleted")
+    apiDeleteCmd.MarkFlagRequired("name")	
+}
+
+func executeDeleteAPICmd(apiname string) {
+
+    err := DeleteAPI(apiname)
+
+    // Result after deleting the API
+    if err == nil {
+        
+        fmt.Println("Successfully removed API")
+        
+    } else {
+        utils.Logln(utils.LogPrefixError+"Deleting API", err)
+    }
+
+    // if flagExportAPICmdToken != "" {
+    //  // token provided with --token (-t) flag
+    //  if exportAPICmdUsername != "" || exportAPICmdPassword != "" {
+    //      // username and/or password provided with -u and/or -p flags
+    //      // Error
+    //      utils.HandleErrorAndExit("username/password provided with OAuth token.", nil)
+    //  } else {
+    //      // token only, proceed with token
+    //  }
+    // } else {
+    //  // no token provided with --token (-t) flag
+    //  // proceed with username and password
+    //  accessToken, apiManagerEndpoint, preCommandErr := utils.ExecutePreCommand(listApisCmdEnvironment, listApisCmdUsername,
+    //      listApisCmdPassword, utils.MainConfigFilePath, utils.EnvKeysAllFilePath)
+
+    //  if preCommandErr == nil {
+    //      if listApisCmdQuery != "" {
+    //          fmt.Println("Search query:", listApisCmdQuery)
+    //      }
+    //      count, apis, err := GetCarbonAppInfo(listApisCmdQuery, accessToken, apiManagerEndpoint)
+
+    //      if err == nil {
+    //          // Printing the list of available APIs
+    //          fmt.Println("Environment:", listApisCmdEnvironment)
+    //          fmt.Println("No. of APIs:", count)
+    //          if count > 0 {
+    //              printAPIs(apis)
+    //          }
+    //      } else {
+    //          utils.Logln(utils.LogPrefixError+"Getting List of APIs", err)
+    //      }
+    //  } else {
+    //      utils.HandleErrorAndExit("Error calling '"+listCmdLiteral+" "+apisCmdLiteral+"'", preCommandErr)
+    //  }
+    // }
+}
+
+// DeleteAPI
+// @param name of the api
+// @return error
+func DeleteAPI(name string) (error) {
+
+    finalUrl := utils.RESTAPIBase + utils.PrefixAPIs + "/" + name
+
+    utils.Logln(utils.LogPrefixInfo+"URL:", finalUrl)
+
+    headers := make(map[string]string)
+    // headers[utils.HeaderAuthorization] = utils.HeaderValueAuthPrefixBearer + " " + accessToken
+
+    resp, err := utils.InvokeDELETERequest(finalUrl, headers)
+
+    if err != nil {
+        utils.HandleErrorAndExit("Unable to connect to "+finalUrl, err)
+    }
+
+    utils.Logln(utils.LogPrefixInfo+"Response:", resp.Status())
+
+    if resp.StatusCode() == http.StatusOK {
+        // apiCarbonAppResponse := &utils.CarbonApp{}
+        // unmarshalError := json.Unmarshal([]byte(resp.Body()), &apiCarbonAppResponse)
+
+        // if unmarshalError != nil {
+        //     utils.HandleErrorAndExit(utils.LogPrefixError+"invalid JSON response", unmarshalError)
+        // }
```

Comment:
Why do we have commented out code here?

Why do we have commented out code here?